---
title: "Master script to run scripts"
author: "Jimmy Zhong"
date: "9/21/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction
Postprandial Hypoglycemic Project

## Data Storage
This is where the data is stored:
```{bash, eval=FALSE}
# Downloaded raw data files
base="/n/data1/joslin/icrb/kostic/data/raw_sequencing_data/GeneWiz_raw_data/Patti_PnF"
batch1="${base}/batch_20210811"
batch2="${base}/batch_20211117"
batch3="${base}/batch_20220906"

# workdir
/n/data1/joslin/icrb/kostic/data/raw_sequencing_data/Patti_PnF_kneadreads

#this is the scratch space being used
/n/scratch3/users/a/adk9/Patti-kneadreads

#this is where post-analysis finalized data output is stored:
/n/data1/joslin/icrb/kostic/ADK/projects/Patti_PnF/
```

## Remove adaptors and host DNA from the metagenome
```{bash, eval=FALSE}
cd ${base}/Patti_PnF-github/quality-control-plus-humann3

batch=batch_20210811
batch=batch_20211117
batch=batch_20220906
./pre-process-reads.sh $batch


cd /n/data1/joslin/icrb/kostic/data/raw_sequencing_data/GeneWiz_raw_data/Patti_PnF/kneaddata

folders=$(ls -d *.gz/)
# folders=$(ls -d */)
for folder in $folders; do
folder=${folder%%"/"}
sample=$(echo $folder | awk -F '_' '{print $1}' | awk -F '-' '{print $1}')
mv $folder $sample
mv ${sample}/*.pair1_kneaddata_paired_1.fastq.gz ${sample}/${sample}.pair1_kneaddata_paired_1.fastq.gz
mv ${sample}/*.pair1_kneaddata_paired_2.fastq.gz ${sample}/${sample}.pair1_kneaddata_paired_2.fastq.gz
done
```


## Run Humann3

Here is where the cleaned samples end up:
${base}/adaptor-trimmed-and-kneaddata-reads

```{bash, eval=FALSE}
num_cores=20
end1=".pair1_kneaddata_paired_1.fastq.gz"
end2=".pair1_kneaddata_paired_2.fastq.gz"
all_samples=$(ls ${base}/adaptor-trimmed-and-kneaddata-reads/*/*${end1})

mkdir -p ${base}/humann3_out

for r1 in $all_samples; do
r2=$(echo $r1 | sed "s/${end1}/${end2}/g")
sample=$(echo $r1 | awk -F'/' '{print $NF}' | sed "s/${end1}//g")
sbatch -p short -c $num_cores --mem=60G -t 0-12:00 /n/data1/joslin/icrb/kostic/jzhong/scripts/run_humann3.bash $sample $r1 $r2 ${base}/humann3_out $num_cores
done

output_dir=${base}/humann3_out/${sample}
humann --input ${output_dir}/${sample}.fastq.gz --output ${output_dir} --threads 20
```
